daemon/state.go

```diff
@@ -281,7 +281,7 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 			// endpoints that don't have a fixed identity or are
 			// not well known.
 			if !identity.IsFixed() && !identity.IsWellKnown() {
-				cache.WaitForInitialIdentities(context.Background())
+				cache.WaitForInitialGlobalIdentities(context.Background())
```

Comment:
shouldn't we check/handle the error here?

Reply:
Only if we pass a context that is canceled (`context.Background()` is not). This is somewhat out of scope of this PR in my opinion.

Reply:
Oh, because these functions only return an error if the context times out, I see.

Yeah, so not for this case but presumably the other cases should pass a timed context?

Agree this could be handled in a separate PR, just seemed weird to me that we ignore an error from a function that could return an error, then keep doing other stuff afterwards, almost as though we assume that this function did not return an error.

Reply:
Filed this is a good first issue: https://github.com/cilium/cilium/issues/8368